core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -867,7 +867,9 @@ protected ObjectListingChunk getObjectListingChunkForPath(String path, boolean r
           while (objectName.startsWith(keyPrefix)) {
             objectName = objectName.substring(0, objectName.lastIndexOf(PATH_SEPARATOR));
             if (!objectName.isEmpty()) {
-              prefixes.add(objectName);
+              // include the separator with the prefix, to conform to what object stores return
+              // as common prefixes.
+              prefixes.add(objectName + PATH_SEPARATOR);
```

Comment:
How do you feel about using `PathUtils` here?

https://github.com/Alluxio/alluxio/blob/master/core/base/src/main/java/alluxio/util/io/PathUtils.java#L288-L290

Reply:
Done